src/services/commercetools/client/createAnonymousClient.ts

```diff
@@ -0,0 +1,22 @@
+import {
+  createApiBuilderFromCtpClient,
+  ByProjectKeyRequestBuilder
+} from '@commercetools/platform-sdk';
+import { Client, ClientBuilder } from '@commercetools/sdk-client-v2';
+import { getEnvVar } from '@/lib/utils';
+import {
+  httpMiddlewareOptions,
+  getAnonymousMiddlewareOptions
+} from '@/services/commercetools/options/options';
+
+const projectKey = getEnvVar(process.env.NEXT_PUBLIC_PROJECT_KEY, 'NEXT_PUBLIC_PROJECT_KEY');
+
+export const createAnonymousClient = (): ByProjectKeyRequestBuilder => {
+  const client: Client = new ClientBuilder()
```

Comment:
Check what official commercetools docs says regarding working with ClientBuilder. Ensure that it's ok creating ClientBuilder instance per each operation. It could happen that creation of ClientBuilder is a costly operation and one instance of ClientBuilder should be used for the whole app.